### PR TITLE
fix: failed to run subquery wrapped in two parentheses

### LINKED ISSUE #791: Physical planner does not support subquery expression
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Datanode

### What happened?

Looks like a datafusion feature missing: https://github.com/apache/arrow-datafusion/blob/master/datafusion/physical-expr/src/planner.rs#L446

Create a table and insert some data:

```sql
CREATE TABLE integers(i INTEGER, j BIGINT TIME INDEX);
INSERT INTO integers VALUES (1, 1), (2, 2), (3, 3), (NULL, 4);
```

Run queries:

```sql
SELECT * FROM integers WHERE i IN ((SELECT i FROM integers)) ORDER BY I;

SELECT * FROM integers WHERE i NOT IN ((SELECT i FROM integers WHERE i=1)) ORDER BY I;

SELECT * FROM integers WHERE i IN ((SELECT i FROM integers)) AND i<3 ORDER BY I;

SELECT i1.i,i2.i FROM integers i1, integers i2 WHERE i IN ((SELECT i FROM integers)) AND i1.i=i2.i ORDER BY 1;
```
All these queries fail with the error:

```text
Failed to execute, error: Datanode { code: 3001, msg: "Failed to execute SQL, 
source: Fail to create physical plan: 
This feature is not implemented: Physical plan does not support logical expression (<subquery>)" }

```

### What operating system did you use?

Mac Osx

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

```bash
Described above.
```

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

fix #791 

also datafusion, arrow and sqlparser got updated

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
LGTM. Good job!

File: tests/cases/distributed/optimizer/filter_push_down.sql [hunk 1]
```diff
@@ -1,15 +0,0 @@
-CREATE TABLE integers(i INTEGER, j BIGINT TIME INDEX);
```
Comment (waynexia) [inline tests/cases/distributed/optimizer/filter_push_down.sql:1]: Why remove this file

Comment (MichaelScofield) [reply tests/cases/distributed/optimizer/filter_push_down.sql:1]: It's moved to the "common" dir, shared by standalone and distributed.

**Review by waynexia — APPROVED** [review]
LGTM

**Pull request merged**